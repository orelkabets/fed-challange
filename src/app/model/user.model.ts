/**
 * Created by or.e on 05/09/2017.
 */
export class UserModel {
    userName: string = '';
    password: string = '';
    userImage: string = './userImage.png';
    active: boolean = false;
    // for dragging positioning
    userNameCords: any = {x:0, y:0};
    userImageCords: any = {x:0, y:0};
}