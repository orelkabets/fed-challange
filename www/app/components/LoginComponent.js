"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var React = require("react");
var user_model_1 = require("../model/user.model");
var update = require("immutability-helper");
var storage_1 = require("../services/storage");
var flexbox_react_1 = require("flexbox-react");
var DashboardComponent_1 = require("./DashboardComponent");
var LoginComponent = /** @class */ (function (_super) {
    __extends(LoginComponent, _super);
    function LoginComponent() {
        var _this = _super.call(this) || this;
        _this.tempUser = null;
        _this.state = { user: null, error: null, isLoggedIn: false };
        return _this;
    }
    LoginComponent.prototype.componentDidMount = function () {
        this.getLoggedInUser();
        if (!this.tempUser)
            this.setState({ user: new user_model_1.UserModel() });
        else {
            this.setState({ user: this.tempUser });
            var userKey = this.tempUser.userName + this.tempUser.password;
            storage_1["default"].updateActive(userKey);
        }
    };
    LoginComponent.prototype.onUserNameChange = function ($event) {
        this.setState({ user: update(this.state.user, { userName: { $set: $event.target.value } }) });
    };
    LoginComponent.prototype.onPasswordChange = function ($event) {
        this.setState({ user: update(this.state.user, { password: { $set: $event.target.value } }) });
    };
    LoginComponent.prototype.onSubmit = function () {
        if (!this.state.user) {
            this.setState({ user: new user_model_1.UserModel(), error: 'Please enter valid user name' });
            return;
        }
        if (!this.state.user.userName.length) {
            this.setState({ error: 'Please enter valid user name' });
            return;
        }
        if (!this.state.user.password.length) {
            this.setState({ error: 'Please enter password!' });
            return;
        }
        var userKey = this.state.user.userName + this.state.user.password;
        var existsUser = storage_1["default"].getItem(userKey);
        if (existsUser) {
            this.setState({ user: existsUser });
        }
        else
            storage_1["default"].setItem(userKey, this.state.user);
        storage_1["default"].updateActive(userKey);
        this.setState({ isLoggedIn: true });
    };
    LoginComponent.prototype.onLogOut = function () {
        storage_1["default"].updateActive(this.state.user.userName + this.state.user.password);
        this.setState({ user: null, isLoggedIn: false, error: null });
        this.setState({ user: new user_model_1.UserModel() });
        this.tempUser = null;
    };
    LoginComponent.prototype.getLoggedInUser = function () {
        var loggedInUser = storage_1["default"].getActiveUser();
        if (loggedInUser) {
            this.setState({ isLoggedIn: true });
            this.tempUser = loggedInUser;
        }
    };
    LoginComponent.prototype.render = function () {
        var _this = this;
        return (this.state.isLoggedIn ?
            React.createElement(flexbox_react_1["default"], { flexDirection: "column" },
                React.createElement(flexbox_react_1["default"], { element: "header", height: "60px", justifyContent: "center" },
                    React.createElement("h4", null,
                        "Welcome ",
                        this.state.user.userName,
                        "!")),
                React.createElement("button", { className: "logout-button", onClick: this.onLogOut.bind(this) }, "Log Out"),
                React.createElement(DashboardComponent_1["default"], { user: this.state.user }))
            :
                React.createElement(flexbox_react_1["default"], { flexDirection: "column", minHeight: "50vh", justifyContent: "center" },
                    React.createElement(flexbox_react_1["default"], { element: "header", height: "60px", justifyContent: "center" },
                        React.createElement("h4", null, "Hello Guest, please login")),
                    React.createElement(flexbox_react_1["default"], { flexDirection: "column", className: "center-component" },
                        React.createElement("input", { id: "user-name", className: "default-input", placeholder: "userName", type: "text", onChange: function ($event) { return _this.onUserNameChange($event); } }),
                        React.createElement("input", { id: "password", className: "default-input", placeholder: "password", type: "password", onChange: function ($event) { return _this.onPasswordChange($event); } }),
                        React.createElement("button", { id: "submit", className: "default-button", onClick: this.onSubmit.bind(this) }, "Let me in."),
                        this.state.error ? React.createElement("span", { className: "error-text" }, this.state.error) : null)));
    };
    return LoginComponent;
}(React.Component));
exports["default"] = LoginComponent;
