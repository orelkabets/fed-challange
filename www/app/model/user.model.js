"use strict";
exports.__esModule = true;
/**
 * Created by or.e on 05/09/2017.
 */
var UserModel = /** @class */ (function () {
    function UserModel() {
        this.userName = '';
        this.password = '';
        this.userImage = './userImage.png';
        this.active = false;
        // for dragging positioning
        this.userNameCords = { x: 0, y: 0 };
        this.userImageCords = { x: 0, y: 0 };
    }
    return UserModel;
}());
exports.UserModel = UserModel;
