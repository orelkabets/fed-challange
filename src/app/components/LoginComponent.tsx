
import * as React from 'react';
import {UserModel} from "../model/user.model";
import * as update   from 'immutability-helper'
import StorageService from "../services/storage";
import Flexbox from 'flexbox-react';
import DashboardComponent from "./DashboardComponent";


interface LoginState {
    user: UserModel;
    error: string;
    isLoggedIn:boolean;
}

export default class LoginComponent extends React.Component<{}, LoginState> {
    tempUser:UserModel = null;
    constructor() {
        super();
        this.state = { user: null , error: null, isLoggedIn: false};
    }

    componentDidMount() {
        this.getLoggedInUser();
        if (!this.tempUser)
            this.setState({ user: new UserModel() });
        else {
            this.setState({user: this.tempUser});
            let userKey = this.tempUser.userName + this.tempUser.password;
            StorageService.updateActive(userKey);
        }

    }

    onUserNameChange($event:any) {
        this.setState({user: update(this.state.user, {userName: {$set: $event.target.value}})});
    }

    onPasswordChange($event:any) {
        this.setState({user: update(this.state.user, {password: {$set: $event.target.value}})});
    }

    onSubmit() {
        if (!this.state.user){
            this.setState({user: new UserModel(), error: 'Please enter valid user name'});
            return;
        }
        if (!this.state.user.userName.length){
            this.setState({error: 'Please enter valid user name'});
            return;
        }
        if (!this.state.user.password.length) {
            this.setState({error: 'Please enter password!'});
            return;
        }
        let userKey:string = this.state.user.userName + this.state.user.password;
        let existsUser:UserModel = StorageService.getItem(userKey);
        if (existsUser) {
            this.setState({user: existsUser});
        } else
            StorageService.setItem(userKey, this.state.user);
        StorageService.updateActive(userKey);
        this.setState({isLoggedIn:true});
    }

    onLogOut() {
        StorageService.updateActive(this.state.user.userName + this.state.user.password);
        this.setState({user:null, isLoggedIn:false, error: null});
        this.setState({user:new UserModel()});
        this.tempUser = null;
    }

    getLoggedInUser() {
        let loggedInUser:UserModel = StorageService.getActiveUser();
        if (loggedInUser) {
            this.setState({isLoggedIn:true});
            this.tempUser = loggedInUser;
        }
    }

    render() {
        return (
        this.state.isLoggedIn ?
            <Flexbox flexDirection="column" >
                <Flexbox element="header" height="60px" justifyContent="center">
                    <h4>Welcome {this.state.user.userName}!</h4>
                </Flexbox>
                <button className="logout-button" onClick={this.onLogOut.bind(this)}>Log Out</button>
                <DashboardComponent user={this.state.user}/>
            </Flexbox>
            :
         <Flexbox flexDirection="column" minHeight="50vh" justifyContent="center">
            <Flexbox element="header" height="60px" justifyContent="center">
                <h4>Hello Guest, please login</h4>
            </Flexbox>

            <Flexbox flexDirection="column" className="center-component" >
                <input id="user-name" className="default-input" placeholder="userName" type="text" onChange={($event:any) => this.onUserNameChange($event)}/>
                <input id ="password" className="default-input" placeholder="password"  type="password" onChange={($event:any) => this.onPasswordChange($event)}/>
                <button id="submit" className="default-button" onClick={this.onSubmit.bind(this)}>Let me in.</button>
                {
                    this.state.error ? <span className="error-text">{this.state.error}</span> : null
                }
            </Flexbox>

        </Flexbox>);
    }
}
