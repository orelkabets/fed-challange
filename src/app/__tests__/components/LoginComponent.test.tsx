import * as React from 'react';
import { create } from 'react-test-renderer';
import {shallow} from 'enzyme';
import LoginComponent from 'LoginComponent';
import {UserModel} from "../../model/user.model";


describe('LoginComponent', () => {
    const fakeUser:UserModel = {
        userName: 'FOO',
        password: 'BAZ',
        userImage: './userImage.png',
        active: false,
        userNameCords:{x:0, y:0},
        userImageCords:{x:0, y:0}
    };
    const invalidUser:UserModel = {
        userName: 'FOO',
        password: '',
        userImage: './userImage.png',
        active: false,
        userNameCords:{x:0, y:0},
        userImageCords:{x:0, y:0}
    };

    it('renders without crashing', () => {
        const tree = create(<LoginComponent/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Successful Login', ()=> {
        const component = shallow(
            <LoginComponent/>
        );
        component.setState({user: fakeUser});
        component.find('#submit').simulate('click');
        let storedUser:UserModel = JSON.parse(localStorage.getItem(fakeUser.userName+fakeUser.password));
        expect(storedUser.userName).toEqual(fakeUser.userName);
    });

    it('Login Failure', ()=> {
        const component = shallow(
            <LoginComponent/>
        );
        component.setState({user: invalidUser});
        component.find('#submit').simulate('click');
        expect(localStorage.getItem(invalidUser.userName+invalidUser.password)).toBe(null);
    });
});