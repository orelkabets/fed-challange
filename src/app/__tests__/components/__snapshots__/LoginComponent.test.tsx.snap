// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent renders without crashing 1`] = `
<div
  className="sc-bdVaJa caNysd"
>
  <header
    className="sc-bdVaJa boYivU"
  >
    <h4>
      Hello Guest, please login
    </h4>
  </header>
  <div
    className="center-component sc-bdVaJa hgkmdP"
  >
    <input
      className="default-input"
      id="user-name"
      onChange={[Function]}
      placeholder="userName"
      type="text"
    />
    <input
      className="default-input"
      id="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
    <button
      className="default-button"
      id="submit"
      onClick={[Function]}
    >
      Let me in.
    </button>
  </div>
</div>
`;
